testapi.pm

```diff
@@ -272,7 +272,7 @@ sub _check_backend_response {
             }
         }
         if (!$check && !$rsp->{saveresult}) {
-            OpenQA::Exception::FailedNeedle->throw(error => "needle(s) '$mustmatch' not found", tags => $mustmatch);
+            OpenQA::Exception::FailedNeedle->throw(error => "unable to match any needles with the tag(s) '$mustmatch'", tags => $mustmatch);
```

Comment:
`no candidate needle with tag(s) '$mustmatch' matched` (discussed with @nicksinger in person)